src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -15684,21 +15684,13 @@ void Unit::ProcDamageAndSpellFor(bool isVictim, Unit* target, uint32 procFlag, u
 
         // Update skills here for players
         // only when you are not fighting other players or their pets/totems (pvp)
-        if (GetTypeId() == TYPEID_PLAYER &&
-                target->GetTypeId() != TYPEID_PLAYER &&
-                !(target->IsTotem() && target->ToTotem()->GetOwner()->IsPlayer()) &&
-                !target->IsPet()
-           )
-        {
-            // On melee based hit/miss/resist/parry/dodge need to update skill (for victim and attacker)
-            if (procExtra & (PROC_EX_NORMAL_HIT | PROC_EX_MISS | PROC_EX_RESIST | PROC_EX_PARRY | PROC_EX_DODGE))
+        if (IsPlayer() && !target->IsCharmedOwnedByPlayerOrPlayer())
```

Comment:
And what about charmed creature?

Reply:
`!target->IsCharmedOwnedByPlayerOrPlayer()` should prevent any skill ups for targets that are players or any creatures owned by a player.
-> No weapon skill during pvp.

Reply:
`IsCharmedOwnedByPlayerOrPlayer` means also charmed creature, not only owned, e.g. using Mind Control on some mob. The previous check did not prevent from that.

Reply:
I can't prove that skill increase should not work on mobs that are temporarily owned by players.
I was using `!IsCharmedOwnedByPlayerOrPlayer()` becasue it makes the code look cleaner (also vmangos is using this check).

Reply:
Okay.